lib/rag/evaluation.ex

```diff
@@ -16,15 +16,20 @@ defmodule Rag.Evaluation do
 
   Prompts from https://github.com/truera/trulens/blob/main/src/feedback/trulens/feedback/prompts.py
   """
-  @spec evaluate_rag_triad(Generation.t(), response_function() | provider()) :: Generation.t()
+  @spec evaluate_rag_triad(Generation.t(), response_function() | provider(), keyword()) ::
+          Generation.t()
   def evaluate_rag_triad(%Generation{halted?: true} = generation, _response_function),
     do: generation
 
-  def evaluate_rag_triad(%Generation{} = generation, %provider_module{} = provider) do
+  def evaluate_rag_triad(%Generation{} = generation, %provider_module{} = provider, _opts) do
     evaluate_rag_triad(generation, &provider_module.generate_text(provider, &1, &2))
   end
 
-  def evaluate_rag_triad(%Generation{} = generation, response_function)
+  def evaluate_rag_triad(%Generation{} = generation, response_function, stream: true) do
+    raise "Streaming"
+  end
+
+  def evaluate_rag_triad(%Generation{} = generation, response_function, opts \\ [])
```

Comment:
I think we don't need the `opts` argument in the evaluation functions.
We don't want to stream the evaluation itself for now.
Instead, we must handle a streamed response.

I think we could ensure that the response is a string and not a stream before evaluating:
```suggestion
  def evaluate_rag_triad(%Generation{response: response} = generation, response_function) when is_binary(response) and is_function(response_function, 2)
```

And then add a last function clause below that raises if non of the previous function clauses match
```elixir
 def evaluate_rag_triad(%Generation{} = generation, response_function) do
    raise "Streaming"
  end
```

We can then do the same for the other evaluation function `detect_hallucination/2`